### ISSUE TITLE
Compiling shared FFMpeg fails

### ISSUE THREAD

**Issue by mkver** [jb-alvarado/media-autobuild_suite#517]
I can't build a shared FFMpeg -- neither as a usual [shared build (FFMpeg option 3)](https://0x0.st/g4c.zip) or with [FFMpeg option 5](https://0x0.st/g4l.zip). Building a static build (with the same ffmpeg-options) works.
[Edit]: It seems to be libbluray: Disabling everything except libbluray leads to failure; disabling libbluray doesn't. [Here](https://0x0.st/gJE.zip) is the log.
And even if I disable libbluray and enable sdl2, I don't get a ffplay.exe when making a shared build. The resulting binaries claim to have been build with --disable-sdl2. This doesn't happen for a static build. Should I open another ticket for this?

**Comment by wiiaboo** [comment]
Concerning sdl2, it's because of issues with sdl2 itself or the MinGW builds of it. Since ffplay depends on sdl2, it can't be built with the shared version.
Actually same reason as libbluray failing apparently. Something in the headers/libs of both are screwing up shared builds.
Since I still have no idea how to fix this, I'll simply add libbluray to be always disabled with shared builds.

If you'd like this to be fixed, I'd advise opening an issue in FFmpeg tracker. You can use the configure and make logs as proof.

**Comment by mkver** [comment]
Thanks for everything! Now it works perfectly.
